utils.js

```diff
@@ -9384,41 +9408,47 @@ export function getPipFrozenTree(basePath, reqOrSetupFile, tempVenvDir) {
       }
       const name = t.name.replace(/_/g, "-").toLowerCase();
       const version = t.version;
-      const exclude = ["pip", "setuptools", "wheel"];
-      if (!exclude.includes(name)) {
```

Comment:
This is a breaking change to track all components including excluded. Perhaps we filter and move such components to formulation.